delta/plugins/graph-analytics/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/graph/analytics/indexing/GraphAnalyticsSinkSuite.scala

```diff
@@ -0,0 +1,172 @@
+      for {
+        expanded <- io
+        types    <- getTypes(expanded)
+        doc      <- JsonLdDocument.fromExpanded(expanded, findRelationship)
+      } yield {
+        val result = GraphAnalyticsResult.Index(
+          id,
+          1,
+          types.getOrElse(Set.empty),
+          Instant.EPOCH,
+          Anonymous,
+          Instant.EPOCH,
+          Anonymous,
+          doc
+        )
+        SuccessElem(Resources.entityType, id, Some(project), Instant.EPOCH, Offset.start, result, 1)
+      }
+    }
+
+    for {
+      r1        <- toIndex(resource1, expanded1)
+      r2        <- toIndex(resource2, expanded2)
+      r3         = SuccessElem(
+                     Resources.entityType,
+                     resource3,
+                     Some(project),
+                     Instant.EPOCH,
+                     Offset.start,
+                     GraphAnalyticsResult.Noop,
+                     1
+                   )
+      chunk      = Chunk.seq(List(r1, r2, r3))
+      // We expect no error
+      _         <- sink(chunk).assert(chunk.map(_.void))
+      // Elasticsearch should have taken into account the bulk query
+      _         <- client.count(index.value).eventually(2L)
+      expected1 <- ioJsonContentOf("result/resource1.json")
+      expected2 <- ioJsonContentOf("result/resource2.json")
+      _         <- client.getSource[Json](index, resource1.toString).eventually(expected1)
+      _         <- client.getSource[Json](index, resource2.toString).eventually(expected2)
+    } yield ()
+
+  }
+
+  test("Push update by query result results") {
+    val chunk = Chunk.seq(
+      List(
+        SuccessElem(
+          Resources.entityType,
+          resource3,
+          Some(project),
+          Instant.EPOCH,
+          Offset.start,
+          GraphAnalyticsResult.UpdateByQuery(file1, Set(nxvFile)),
+          1
+        ),
+        SuccessElem(
+          Resources.entityType,
+          resource3,
+          Some(project),
+          Instant.EPOCH,
+          Offset.start,
+          GraphAnalyticsResult.Noop,
+          1
+        ),
+        FailedElem(
+          Resources.entityType,
+          resource3,
+          Some(project),
+          Instant.EPOCH,
+          Offset.start,
+          new IllegalStateException("BOOM"),
+          1
+        )
+      )
+    )
+
+    for {
+      _         <- sink(chunk).assert(chunk.map(_.void))
+      // The reference to file1 should have been resolved and introduced as a relationship
+      // The update query should not have an effect on the other resource
+      expected1 <- ioJsonContentOf("result/resource1_updated.json")
+      expected2 <- ioJsonContentOf("result/resource2.json")
+      _         <- client.count(index.value).eventually(2L)
+      _         <- client.getSource[Json](index, resource1.toString).eventually(expected1)
+      _         <- client.getSource[Json](index, resource2.toString).eventually(expected2)
+    } yield ()
```

Comment:
I think here it's probably helpful to clarify in variable names what things are doing. If we called `resource3` `resourceWithRelationshipToResource1` then it's much more understandable what the test is doing.

`resource1_updated.json` could become `resource1_with_extra_relationship`
(or, `expected1` could become `resource1WithExtraRelationship`

Even better, we could say `client.getSource[Json](index, resource1.toString) should haveRelationshipTo(resource3)`. This would really spell out what the test is about

Another point, I don't think it's necessary to have negative assertions. And whenever negative assertions are in an eventually block, they might not work anyway